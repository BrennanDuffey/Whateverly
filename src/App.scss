.App {
  text-align: center;
}

main {
  margin-top: 175px;
}





