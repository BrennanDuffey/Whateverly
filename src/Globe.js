import React, { Component } from 'react';
import Button from './Button.js';
import AnimalCard from './AnimalCard.js';
import CountryCard from './CountryCard.js'
// import worldMap from './worldmap.jpg'


class Globe extends Component {
  constructor(props) {
    super(props);
    this.state = {}
  }
  
  renderSpecies = () => {
    return this.props.endangeredSpecies
    .filter(species => species.name === this.props.searchInput)
    .map(species => {
      return <AnimalCard {...species} key={species.name}/>
    })
  }

  renderCountry = () => {
    return this.props.countries.filter(country => country.name === this.props.searchInput).map(country => {
      return <CountryCard {...country}  key={country.name}/>
    })
  }

  renderCountrySpecies = () => {
    return this.props.endangeredSpecies
    .filter(species => species.locations.includes(this.props.searchInput))
    .map(species => {
      return <AnimalCard {...species} key={species.name}/>
    })
  }


  render() {
    return(
      <main>
        {
          this.renderSpecies(),
          this.renderCountry()
          // this.renderCountrySpecies()
        }
        {
          this.renderCountrySpecies()
        }
      </main>
    )
  }
}




export default Globe;
{/* <picture>
  {/* <img src={worldMap} alt="World Map" /> 
</picture>
<Button /> */}