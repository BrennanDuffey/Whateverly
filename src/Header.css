header {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 0px 190px;
  margin: 10px;
}

form {
  display: flex;
  padding-top: 28px;
}

input {
  -webkit-appearance: none;
  font-size: 1rem;
  height: 28px;
  width: 54%;
}

button {
  height: 27px;
}