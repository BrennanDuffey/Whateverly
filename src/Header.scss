header {
  background: rgb(53, 53, 53);
  box-shadow: 5px 15px 7px rgba(0, 0, 0, .5);
  color: #FCFDFF;
  display: flex;
  flex-direction: column;
  align-items: center;
  position: fixed;
  top: 0;
  width: 100%;
  padding: 10px 0;
}

form {
  display: flex;
  padding-top: 28px;
}

.app-title {
  font-size: 3rem;
  margin: 0;
  padding: 0;
}

input {
  -webkit-appearance: none;
  font-size: 1rem;
  height: 28px;
  width: 50%;
  margin: 10px 0;
}

.show-btn {
  background: #102731;
  border-style: none;
  color: #fff;
  height: 28px;
  width: 50%;
}

.show-btn:hover {
  background: #8FA7AD;
}
