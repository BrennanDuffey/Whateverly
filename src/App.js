import React, { Component } from 'react';
import Globe from './Globe.js';
import Header from './Header.js';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      endangeredSpecies: [],
      countries: [],
      searchInput: ''
    }
  }

  componentDidMount = () => {
    fetch('https://fe-apps.herokuapp.com/api/v1/whateverly/1901/KaylaLawson/endangeredSpecies')
    .then(response => response.json())
    .then(data => this.setState({endangeredSpecies: data.endangeredSpecies }))
    .catch(err => console.log(err))

    fetch('https://fe-apps.herokuapp.com/api/v1/whateverly/1901/KaylaLawson/countries')
    .then(response => response.json())
    .then(data => this.setState({countries: data.countries}))
    .catch(err => console.log( err))
  }
  
  generateContent = (inputValue) => {
    // console.log(inputValue)
    this.setState({
      searchInput: inputValue
    })

  }

  render() {
    // console.log(this.state.endangeredSpecies);
    console.log(this.state.countries)
    return (
      <div className="App">
       <Header generateContent={this.generateContent} /> 
       <Globe searchInput={this.state.searchInput} 
            endangeredSpecies={this.state.endangeredSpecies} 
            countries={this.state.countries} /> 
      </div>
    );
  }
}

export default App;
