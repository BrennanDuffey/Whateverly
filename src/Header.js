import React, { Component } from 'react';
import './Header.css';

class Header extends Component {
  constructor() {
    super();
    this.state = {
      showFavorites: true
    }
  }

  handleChange = (e) => {
    e.preventDefault();
    this.props.generateContent(e.target.value.trim().toLowerCase());
  }

  handleClick = (e) => {
    e.preventDefault();
    this.props.setFavoriteState(this.state.showFavorites)
    this.setState({showFavorites: !this.state.showFavorites})
  }

  render() {
    let showFavBtn = 'Show Favorites';
    if (!this.state.showFavorites) {
      showFavBtn = "Show All";
    } 

    return(
      <header>
        <h1>Kayla's Ark</h1>
        <input type="search" placeholder=" Search Animal or Country..." onChange={this.handleChange} />  
        <button onClick={this.handleClick}>{showFavBtn}</button>
      </header>
    
    )
  }
}

export default Header;
