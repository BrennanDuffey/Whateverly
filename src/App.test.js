import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App.js';
import Globe from './Globe.js';
import Header from './Header.js';


describe('App', () => {
  
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( 
      <App/>
    );
  });

  it('Should render correctly with all the information passed', () => {
      expect(wrapper).toMatchSnapshot();
  });

  it('Should have default states', () => {
    expect(wrapper.state()).toEqual({
      endangeredSpecies: [],
      countries: [],
      searchInput: '',
      favAnimals: [],
      isFavorite: false
    });
  });

  it('Should be able to update favAnimals state', () => {
    expect(wrapper.state().favAnimals).toEqual([]);
    wrapper.instance().addFav('Jaguar');
    expect(wrapper.state().favAnimals).toEqual(['Jaguar']);
  });

  it('Should be able to update favAnimals state', () => {
    expect(wrapper.state().favAnimals).toEqual(['Jaguar']);
    wrapper.instance().addFav('Jaguar');
    expect(wrapper.state().favAnimals).toEqual([]);
  });

  it('Should be able to update searchInput state', () => {
    expect(wrapper.state().searchInput).toEqual('');
    wrapper.instance().generateContent('Jaguar');
    expect(wrapper.state().searchInput).toEqual('Jaguar');
  });


  it('Should be able to update isFavorite state', () => {
    expect(wrapper.state().isFavorite).toEqual(false);
    wrapper.instance().setFavoriteState(true);
    expect(wrapper.state().isFavorite).toEqual(true);
  });
});

