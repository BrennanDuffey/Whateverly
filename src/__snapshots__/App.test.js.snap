// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render correctly with all the information passed 1`] = `
<div
  className="App"
>
  <Header
    generateContent={[Function]}
    setFavoriteState={[Function]}
  />
  <Globe
    addFav={[Function]}
    countries={Array []}
    endangeredSpecies={Array []}
    favAnimals={Array []}
    isFavorite={false}
    searchInput=""
  />
</div>
`;
